urllib3/poolmanager.py

```diff
@@ -148,8 +153,14 @@ class PoolManager(RequestMethods):
 
     proxy = None
 
-    def __init__(self, num_pools=10, headers=None, **connection_pool_kw):
+    def __init__(self, num_pools=10, headers=None, hijacked_dns_resolver=None,
+                 **connection_pool_kw):
         RequestMethods.__init__(self, headers)
+
+        self.hijacked_dns_resolver = hijacked_dns_resolver
+        if not isinstance(self.hijacked_dns_resolver, dict):
```

Comment:
Let's not do this. Anything that implements `Mapping` is fine, and in fact we shouldn't test for type at all. Let's just do `if hijacked_dns_resolver is None:` instead.